mesonpy/__init__.py

```diff
@@ -995,6 +996,34 @@ def _project(config_settings: Optional[Dict[Any, Any]]) -> Iterator[Project]:
         yield project
 
 
+def _get_ninja_if_needed() -> List[str]:
+    env_ninja = os.environ.get('NINJA', None)
+    ninja_candidates = [env_ninja] if env_ninja else ['ninja', 'ninja-build', 'samu']
+    for ninja in ninja_candidates:
+        ninja_path = shutil.which(ninja)
+        if ninja_path is None:
+            continue
+
+        result = subprocess.run([ninja_path, '--version'], check=False, text=True, capture_output=True)
+
+        try:
+            candidate_version = tuple(int(x) for x in result.stdout.split('.')[:3])
+        except ValueError:
+            # The meson search function skips forward if the version is not readable or too low
+            continue
+        if candidate_version < (1, 8, 2):
```

Comment:
I would have defines the minimum version as a module variable and used it where you need it. This is the second place where it used and in this cases it is inevitable that some day someone will update one but not the other. I would do something like:
```python
# up where other constants are defined
_NINJA_REQUIRED_VERSION = (1, 8, 2)
# in the _depstr class definition
    ninja = 'ninja >= {}'.format('.'.join(_NINJA_REQUIRED_VERSION))
# here
if candidate_version < _NINJA_REQUIRED_VERSION
```
or something along these linesm (also see above about the function argument).